app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      login(params[:email], params[:password])
+      flash[:success] = "Welcome! You have signed up successfully"
+      redirect_to root_path
+    else
+      if @user.errors.messages[:password] != []
+        flash[:error] = "Password has to be minimum 6 caracters"
+      elsif @user.errors.messages[:password_confirmation]
+        flash[:error] = "Account not created, passwords don't match"
```

Comment:
Why a flash for these password errors- isnt' it enough to display them along with other model errors on the edit-view?

Reply:
So true